tests/testthat/test-add_co2-summarize_co2_range.R

```diff
@@ -0,0 +1,117 @@
+  expect_equal(nrow(filter(out, benchmark == .env$benchmark)), expected)
+
+  benchmark <- "unit"
+  expected <- 6
+  # 6 = 3 emission_profile * 2 unit
+  out <- summarize_co2_range(data)
+  expect_equal(nrow(filter(out, benchmark == .env$benchmark)), expected)
+
+  benchmark <- "tilt_sector"
+  expected <- 12
+  # 12 = 3 emission_profile * 2 tilt_sector * 2 tilt_subsector
+  out <- summarize_co2_range(data)
+  expect_equal(nrow(filter(out, benchmark == .env$benchmark)), expected)
+
+  benchmark <- "unit_tilt_sector"
+  expected <- 24
+  # 24 = 3 emission_profile * 2 tilt_sector * 2 tilt_subsector * 2 unit
+  out <- summarize_co2_range(data)
+  expect_equal(nrow(filter(out, benchmark == .env$benchmark)), expected)
+})
+
+test_that("with a simple case yields the same as `summarize_range()` (214#issuecomment-2061180499)", {
+  # styler: off
+  data <- tribble(
+    ~benchmark, ~emission_profile, ~co2_footprint, ~unit, ~tilt_sector, ~tilt_subsector, ~isic_4digit,
+         "all",             "low",             1L,  "m2",    "sector1",    "subsector1",     "'1234'",
+         "all",          "medium",             2L,  "m2",    "sector1",    "subsector2",     "'1234'"
+  )
+  # styler: on
+
+  expect_equal(
+    summarize_range(
+      data,
+      col_footprint(),
+      .by = c(col_grouped_by(), col_risk_category_emissions_profile())
+    ),
+    summarize_co2_range(data)
+  )
+})
+
+test_that("is vectorized over `benchmark`", {
+  # styler: off
+  data <- tribble(
+    ~benchmark, ~emission_profile, ~co2_footprint, ~unit, ~tilt_sector, ~tilt_subsector, ~isic_4digit,
+         "all",             "low",             1L,  "m2",    "sector1",    "subsector1",     "'1234'",
+        "unit",             "low",             1L,  "m2",    "sector1",    "subsector1",     "'1234'",
+  )
+  # styler: on
+
+  out <- summarize_co2_range(data)
+  expect_equal(unique(out$benchmark), c("all", "unit"))
+})
+
+test_that("without crucial columns errors gracefully", {
+  # styler: off
+  data <- tribble(
+          ~benchmark, ~emission_profile, ~co2_footprint, ~unit, ~tilt_sector, ~tilt_subsector, ~isic_4digit,
+               "all",             "low",             1L,  "m2",    "sector1",    "subsector1",     "'1234'",
+              "unit",             "low",             1L,  "m2",    "sector1",    "subsector1",     "'1234'",
+       "tilt_sector",             "low",             1L,  "m2",    "sector1",    "subsector1",     "'1234'",
+    "tilt_subsector",             "low",             1L,  "m2",    "sector1",    "subsector1",     "'1234'",
+       "isic_4digit",             "low",             1L,  "m2",    "sector1",    "subsector1",     "'1234'",
+  )
+  # styler: on
+
+  crucial <- col_footprint()
+  bad <- select(data, -all_of(crucial))
+  expect_error(summarize_co2_range(bad), crucial)
+
+  crucial <- col_grouped_by()
+  bad <- select(data, -all_of(crucial))
+  expect_error(summarize_co2_range(bad), class = "check_matches_name")
```

Comment:
@maurolepore What's the origin of class `check_matches_name`, and why it's only used for columns "col_grouped_by" and "emission_profile" ?

Reply:
Makes sense!